AmazonSQS/AmazonSQSJCAAPI/src/main/java/fish/payara/cloud/connectors/amazonsqs/api/AwsCredentialsProviderUtils.java

```diff
@@ -0,0 +1,52 @@
+package fish.payara.cloud.connectors.amazonsqs.api;
+
+import com.amazonaws.auth.*;
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+import com.amazonaws.util.StringUtils;
+
+/**
+ * @author Martin Charlesworth
+ */
+public class AwsCredentialsProviderUtils {
+
+	/**
+	 * Creates the appropriate type of {@link AWSCredentialsProvider} depending on the configuration provided.
+	 */
+	public static AWSCredentialsProvider getProvider(String accessKey,
+													 String secretKey,
+													 String sessionToken,
+													 String profileName,
+													 boolean useIAMRole) {
+		AWSCredentialsProvider awsCredentialsProvider = null;
```

Comment:
I think you bring an interesting issue here. but you don't need all the additional stuff to trigger the EC2ContainerCredentialsProviderWrapper. I would remove the useIAMRole flag entirely. Then update this "getProviderMethod()" to default the awsCredentialsProvider to DefaultAWSCredentialsProviderChain.getInstance(). That way if profileName and access/secret keys are null, it will use the Default Chain, which has your EC2ContainerCredentialsProviderWrapper. This Chain is precisely what things like lambdas and EC2 instances use when they are configuring a client.

`
public class DefaultAWSCredentialsProviderChain extends AWSCredentialsProviderChain {

    private static final DefaultAWSCredentialsProviderChain INSTANCE
        = new DefaultAWSCredentialsProviderChain();

    public DefaultAWSCredentialsProviderChain() {
        super(new EnvironmentVariableCredentialsProvider(),
              new SystemPropertiesCredentialsProvider(),
              WebIdentityTokenCredentialsProvider.create(),
              new ProfileCredentialsProvider(),
              new EC2ContainerCredentialsProviderWrapper());
    }

    public static DefaultAWSCredentialsProviderChain getInstance() {
        return INSTANCE;
    }
}
`

Reply:
thanks Ken, I'll take a look into that